drivers/syscon/Kconfig

```diff
@@ -3,33 +3,31 @@
 # Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
 # SPDX-License-Identifier: Apache-2.0
 
-#
-# SYSCON options
-#
-menuconfig SYSCON
-	bool "SYSCON (System Controller) drivers"
+# SYSCON (System Controller) drivers. System controller node represents
+# a register region containing a set of miscellaneous registers. The
+# registers are not cohesive enough to represent as any specific type
+# of device. The typical use-case is for some other node's driver, or
+# platform-specific code, to acquire a reference to the syscon node and
+# extract information from there.
+
+DT_COMPAT_SYSCON_GENERIC := syscon
+
+config SYSCON_GENERIC
```

Comment:
With this change, the syscon generic driver is suddenly present at the top-level drivers Kconfig and not in a submenu like most other drivers.

![image](https://user-images.githubusercontent.com/9802583/134904022-39b91b17-1310-42b8-9bdc-f59b40789172.png)

The drivers menu is already very big, so please get this into a submenu again.